tests/Doctrine/Tests/DBAL/Query/QueryBuilderTest.php

```diff
@@ -863,4 +863,23 @@ public function testGetParameterTypes()
 
         $this->assertSame(array('name' => \PDO::PARAM_STR, 'isActive' => \PDO::PARAM_BOOL), $qb->getParameterTypes());
     }
+
+    /**
+     * @group DBAL-790
+     */
+    public function testSelectWithJoinsWithNotUniqueAliases()
+    {
+        $qb = new QueryBuilder($this->conn);
+
+        $qb->select('a.id, b.id')
+            ->from('table_a', 'a')
+            ->join('a', 'table_b', 'b', 'a.fk_b = b.id')
+            ->join('a', 'table_b', 'b', 'a.fk_b = b.id')
+            ->join('b', 'table_a', 'a', 'a.fk_b = b.id');
+
+        $this->assertEquals(
+            'dumb string',
+            (string) $qb
+        ); //you will get out of memory anyway
```

Comment:
Would be best to add the test to your patch PR and assert that the `QueryException`  is thrown.